extensions/modules/src/org/exist/xquery/modules/mail/SendEmailFunction.java

```diff
@@ -343,22 +344,17 @@ public SMTPException(Throwable cause)
         final int TCP_PROTOCOL_SMTP = 25;   //SMTP Protocol
         String smtpResult = "";             //Holds the server Result code when an SMTP Command is executed
 
-        Socket smtpSock = null;
-        BufferedReader smtpIn = null;
-        PrintWriter smtpOut = null;
-
         List<Boolean> sendMailResults = new ArrayList<>();
 
-        try
-        {
-            //Create a Socket and connect to the SMTP Server
-            smtpSock = new Socket(SMTPServer, TCP_PROTOCOL_SMTP);
+        try (//Create a Socket and connect to the SMTP Server
+             Socket smtpSock = new Socket(SMTPServer, TCP_PROTOCOL_SMTP);
+
+             //Create a Buffered Reader for the Socket
+             BufferedReader smtpIn = new BufferedReader(new InputStreamReader(smtpSock.getInputStream()));
```

Comment:
`smtpSock.getInputStream` should be bound to it's own variable before the `BufferedReader` to ensure that it is actually closed. Same with the `OutputStream`:

e.g.
```java
try (
  final InputStream is = smtpSock.getInputStream();
  final Reader smtpIn = new BufferedReader(new InputStreamReader(is));
  final OutputStream os = smtpSock.getOutputStream();
  final OutputStreamWriter osw = new OutputStreamWriter(os, charset)
  final Writer smtpOut = new PrintWriter(osw)
) {
```

Reply:
is this for style, or does the JVM/compiler not recognise the original way of coding? (wondering what the spec is)